### PR TITLE
Loss function updates

### PR DESCRIPTION
**Pull request by Adamits**
- Changes loss functions to use `CrossEntropyLoss`.
- Overrides this for `PointerGenerator`, which needs to use `NLLLoss`
- Patches the main bug in #111

### CODE REVIEW

File: yoyodyne/models/pointer_generator.py [hunk 1]
```diff
@@ -358,18 +358,75 @@ def forward(
...
-                    source_encoded,
-                    batch.source.mask,
-                    batch.source.padded,
-                    last_hiddens,
-                    self.teacher_forcing if self.training else False,
-                    features_enc=features_encoded,
-                    features_mask=batch.features.mask,
-                    target=batch.target.padded if batch.target else None,
-                )
+            predictions = self.decode(
```
Comment (kylebgorman) [inline yoyodyne/models/pointer_generator.py:361]: Is this a fix of sorts for #111?

File: yoyodyne/models/pointer_generator.py [hunk 2]
```diff
@@ -358,18 +358,75 @@ def forward(
...
+                features_mask=batch.features.mask,
+                target=batch.target.padded if batch.target else None,
+            )
         return predictions
 
+    def _get_loss_func(
+        self,
+    ) -> Callable[[torch.Tensor, torch.Tensor], torch.Tensor]:
+        """Returns the actual function used to compute loss.
+
```
Comment (kylebgorman) [inline yoyodyne/models/pointer_generator.py:377]: Could you add some brief documentation here that we're overriding the base version because we need NLLLoss for reasons? It could go into the docstring or even just a comment at line 381-382 or so.

**Review by kylebgorman — APPROVED** [review]
LGTM, suggested slight documentation though.

File: yoyodyne/models/pointer_generator.py (hunk 1 above)
Comment (Adamits) [reply yoyodyne/models/pointer_generator.py:361]: Yes. I should mark it in the PR description
